### PR TITLE
Remove memory_pool

### PR DESCRIPTION
**Pull request by MBkkt**

### CODE REVIEW

File: tests/formats/formats_10_tests.cpp [hunk 1]
```diff
@@ -110,8 +110,8 @@ class format_10_test_case : public tests::format_test_case {
     auto writer =
       codec->get_postings_writer(false, irs::IResourceManager::kNoop);
     ASSERT_NE(nullptr, writer);
-    irs::postings_writer::state
-      term_meta;  // must be destroyed before the writer
+    irs::version10::term_meta term_meta;
+    // must be destroyed before the writer
```
Comment (Dronplane) [inline tests/formats/formats_10_tests.cpp:114]:
```suggestion

```
Comment seems obsolete

**Review by Dronplane — APPROVED** [review]
LGTM

File: tests/formats/formats_10_tests.cpp [hunk 2]
```diff
@@ -493,26 +485,22 @@ TEST_P(format_10_test_case, postings_read_write) {
...
+      writer->encode(*out, meta1);
     }
 
     // check doc positions for term0 & term1
-    {
-      auto& typed_meta0 = static_cast<irs::version10::term_meta&>(*meta0);
-      auto& typed_meta1 = static_cast<irs::version10::term_meta&>(*meta1);
-      ASSERT_GT(typed_meta1.doc_start, typed_meta0.doc_start);
-    }
+    ASSERT_GT(meta0.doc_start, meta1.doc_start);
```
Comment (MBkkt) [inline tests/formats/formats_10_tests.cpp:503]:
```suggestion
    ASSERT_LT(meta0.doc_start, meta1.doc_start);
```

File: core/formats/formats_burst_trie.cpp [hunk 3]
```diff
@@ -402,12 +400,12 @@ class entry : private util::noncopyable {
...
 
-  const irs::postings_writer::state& term() const noexcept {
-    return *mem_.as<irs::postings_writer::state>();
+  const version10::term_meta& term() const noexcept {
+    return *mem_.as<version10::term_meta>();
   }
 
-  irs::postings_writer::state& term() noexcept {
-    return *mem_.as<irs::postings_writer::state>();
+  const version10::term_meta& term() noexcept {
```
Comment (MBkkt) [inline core/formats/formats_burst_trie.cpp:407]:
```suggestion
  version10::term_meta& term() noexcept {
```

**Pull request merged**